File: e003.c
#include <stdio.h>
#include <math.h>

//int main(void)
int e3(long num)
{
    //long num = 600851475143;
    int i = floor(sqrt(num / 2));
    int j = 0;
    if (i % 2 == 0)
    {
        i -= 1;
    }

    int mxp = 0;

    while (i > 0)
    {
        j = 2;
        if (num % i == 0)
        {
            //printf("%d\n", i);
            int counter = 0;
            while (j <= i / 2)
            {
                if (i % j == 0)
                {
                    counter++;
                }
                j++;
            
            }
            if (counter == 0)
            {
                mxp = i;
                break;
            }
        }
        i -= 2;
    }
    printf("%d\n", mxp);
    return (0);
}



// driver

int main(void)
{
    int n = 13195;
    e3(n);
    
    long num = 600851475143;
    e3(num);

    return (0);

}

File: e003.java
//import java.lang.Math;

public class Gg 
{
    static void drei(long n)
    {
        int x = (int) Math.sqrt(n / 2);
        int y, p;
        y = p = 0;
        int count;
    
        if (x % 2 == 0)
            x--; 

        while (x > 0)
        {
            y = 2;
            if (n % x == 0)
            {
                count = 0;
            
                while (y <= x / 2)
                {
                    if (x % y == 0)
                    {
                        count++;
                    }
                    y++;
                }
            
                if (count == 0)
                {
                    p = x;
                    break;
                }
            }
            x -= 2;
        }
    
        System.out.println(p);
    }


    public static void main(String args[])
    {
        long n = 13195;
        drei(n); 
        
        n  =  600851475143l;
        drei(n);
    }
}
